packages/nimble-components/src/table-column/icon/cell-view/index.ts

```diff
@@ -0,0 +1,80 @@
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { ViewTemplate, observable } from '@microsoft/fast-element';
+import { TableCellView } from '../../base/cell-view';
+import { template } from './template';
+import type {
+    TableColumnEnumCellRecord,
+    TableColumnEnumColumnConfig
+} from '../../enum-base';
+import {
+    IconView,
+    MappingIconConfig
+} from '../../enum-base/models/mapping-icon-config';
+import type { IconSeverity } from '../../../icon-base/types';
+import { MappingSpinnerConfig } from '../../enum-base/models/mapping-spinner-config';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-column-icon-cell-view': TableColumnIconCellView;
+    }
+}
+
+/**
+ * The cell view for the icon column
+ */
+export class TableColumnIconCellView
+    extends TableCellView<
+    TableColumnEnumCellRecord,
+    TableColumnEnumColumnConfig
+    >
+    implements IconView {
+    @observable
+    public severity: IconSeverity;
+
+    @observable
+    public text?: string;
+
+    @observable
+    public iconTemplate?: ViewTemplate<IconView>;
+
+    @observable
+    public visual?: 'spinner' | 'icon';
+
+    private columnConfigChanged(): void {
+        this.updateState();
+    }
+
+    private cellRecordChanged(): void {
+        this.updateState();
+    }
+
+    private updateState(): void {
+        this.visual = undefined;
+        if (!this.columnConfig?.mappingConfigs || !this.cellRecord) {
```

Comment:
mappingConfigs isn't nullable right?

```suggestion
        if (!this.columnConfig || !this.cellRecord) {
```

Reply:
I had run into a situation where we were accessing `mappingConfigs` when it wasn't defined, but I can't seem to reproduce now (in Storybook or in unit tests). I guess I'll just make that change.